html/browsers/browsing-the-web/navigating-across-documents/008.html

```diff
@@ -1,17 +1,37 @@
 <!doctype html>
-<title>Link with onclick form submit to javascript url and href navigation </title>
+<link rel="help" href="https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigating-across-documents">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/browsing-the-web.html#javascript-protocol">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#concept-form-submit">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#submit-get-action">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#plan-to-navigate">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#planned-navigation">
+<title>Link with onclick form submit to javascript: url is aborted by href navigation</title>
 <script src="/resources/testharness.js"></script>
 <script src="/resources/testharnessreport.js"></script>
 <div id="log"></div>
 <iframe id="test" name="test"></iframe>
-<form target="test" action="javascript:'<script>parent.postMessage(&quot;click&quot;, &quot;*&quot;)</script>'"></form>
+<form target="test" action="javascript:parent.postMessage('click', '*')"></form>
```

Comment:
This is changing the test significantly.  The old test tested what happens when a javascript: URI returns a string, which is then treated as the body of a response to navigate to and ends up running scripts itself.  The new test just runs script to post a message.  Those are different operations, and we should probably have tests for both.

It's not actually 100% clear to me from the spec what the behavior should be for this rewritten testcase (nor indeed for the pre-rewrite one)...

Reply:
> This is changing the test significantly. The old test tested what happens when a javascript: URI returns a string, which is then treated as the body of a response to navigate to and ends up running scripts itself. 
The new test just runs script to post a message. Those are different operations, and we should probably have tests for both

Yes, all of this is absolutely true. [Chrome does not honor the target attribute when handling "`javascript:` URL" navigations](https://bugs.chromium.org/p/chromium/issues/detail?id=749492), so running this test results in a blank screen, which is a trash result for wpt. To properly test this, I've created new tests that nest the entire operation into an iframe—those are in a different branch.